packages/codegen/src/generate-descriptors.ts

```diff
@@ -307,36 +307,40 @@ type SeparateUndefined<T> = undefined extends T
   ? undefined | Exclude<T, undefined>
   : T
 
+
 type Anonymize<T> = SeparateUndefined<
-  T extends FixedSizeBinary<infer L> ? FixedSizeBinary<L> :
-  T extends
-    | string
-    | number
-    | bigint
-    | boolean
-    | void
-    | undefined
-    | null
-    | symbol
-    | Binary
-    | Uint8Array
-    | Enum<any>
-    ? T
-    : T extends AnonymousEnum<infer V>
-      ? Enum<V>
-      : T extends MyTuple<any>
-        ? {
-            [K in keyof T]: T[K]
-          }
-        : T extends []
-          ? []
-          : T extends FixedSizeArray<infer L, infer T>
-            ? number extends L
-              ? Array<T>
-              : FixedSizeArray<L, T>
-            : {
-                [K in keyof T & string]: T[K]
-              }
+  T extends FixedSizeBinary<infer L>
+    ? number extends L
+      ? Binary
+      : FixedSizeBinary<L>
+    : T extends
+          | string
+          | number
+          | bigint
+          | boolean
+          | void
+          | undefined
+          | null
+          | symbol
+          | Binary
```

Comment:
Maybe for later, this line is now redundant

Reply:
indeed, thanks!